mutmut/node_mutation.py

```diff
@@ -125,6 +128,62 @@ def operator_string_methods_swap(
+                # split() -> split()                                        # not len(node.args)
+                # rsplit() -> rsplit()                                      # not len(node.args)
+                # split(" ") -> split(" ")                                  # len(node.args) == 1 & maxsplit not in args
+                # rsplit(" ") -> rsplit(" ")                                # len(node.args) == 1 & maxsplit not in args
+                # split(sep=" ") -> split(sep=" ")                          # len(node.args) == 1 & maxsplit not in args
+                # rsplit(sep=" ") -> rsplit(sep=" ")                        # len(node.args) == 1 & maxsplit not in args
+                # split(maxsplit=-1) -> split(maxsplit=-1)                  # maxsplit=-1 in args
+                # rsplit(maxsplit=-1) -> rsplit(maxsplit=-1)                # maxsplit=-1 in args
+                # split(" ", maxsplit=-1) -> split(" ", maxsplit=-1)        # maxsplit=-1 in args
+                # rsplit(" ", maxsplit=-1) -> rsplit(" ", maxsplit=-1)      # maxsplit=-1 in args
+
+                # split(maxsplit=1) -> rsplit(maxsplit=1)                   # maxsplit in args and maxsplit != -1
+                # rsplit(maxsplit=1) -> split(maxsplit=1)                   # maxsplit in args and maxsplit != -1
+
+                # split(" ", 1) -> rsplit(" ", 1)                           # len(node.args) == 2 & maxsplit not in args
+                # rsplit(" ", 1) -> split(" ", 1)                           # len(node.args) == 2 & maxsplit not in args
+
+                # split(" ", maxsplit=1) -> rsplit(" ", maxsplit=1)         # maxsplit in args and maxsplit != -1
+                # rsplit(" ", maxsplit=1) -> split(" ", maxsplit=1)         # maxsplit in args and maxsplit != -1
+                key_args: set[str] = {a.keyword.value for a in node.args if a.keyword}
+
+                maxsplit_val = None
+                for a in node.args:
+                    if a.keyword and a.keyword.value == "maxsplit":
+                        if any([isinstance(child, cst.Minus) for child in a.value.children]):
+                            maxsplit_val = int(a.value.expression.value) * -1
+                            break
+                        if any([isinstance(child, cst.Plus) for child in a.value.children]):
+                            maxsplit_val = int(a.value.expression.value)
+                            break
+                        maxsplit_val = a.value.evaluated_value
+
+
+                if (
+                        not len(node.args) or
+                        (len(node.args) == 1 and "maxsplit" not in key_args) or
+                        ("maxsplit" in key_args and maxsplit_val == -1)
+                ):
+                    continue
+
+                if (
+                        (len(node.args) == 2 and "maxsplit" not in key_args) or
+                        ("maxsplit" in key_args and maxsplit_val != -1)
+                ):
```

Comment:
We could rewrite it with the libcst matchers. I think both ways are quite readable, so choose whatever you prefer.

This would be the option with the libcst methods:

```python
                # either maxsplit is passed as the second arg (so 2+ args)
                if (m.matches(node, m.Call(args=[m.AtLeastN(n=2)]))
                    # or it is passed at any possition as a keyword arg
                    or m.matches(node, m.Call(args=[m.ZeroOrMore(), m.Arg(keyword=m.Name(value="maxsplit")), m.ZeroOrMore()]))):

                    func_name = cst.ensure_type(node.func, cst.Attribute).attr
                    yield node.with_deep_changes(func_name, value=new_call)
```